pkg/kn/commands/service_create.go

```diff
@@ -53,70 +60,193 @@ func NewServiceCreateCommand(p *KnParams) *cobra.Command {
 					return err
 				}
-				fmt.Fprintf(cmd.OutOrStdout(), "Service '%s' successfully created in namespace '%s'.\n", args[0], namespace)
+				return showUrl(client, name, namespace, cmd.OutOrStdout())
 			}
+
 			return nil
 		},
 	}
 	AddNamespaceFlags(serviceCreateCommand.Flags(), false)
 	editFlags.AddCreateFlags(serviceCreateCommand)
 	return serviceCreateCommand
 }
+
+func createService(client serving_v1alpha1_client.ServingV1alpha1Interface, service *serving_v1alpha1_api.Service, namespace string, out io.Writer) error {
+	_, err := client.Services(namespace).Create(service)
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(out, "Service '%s' successfully created in namespace '%s'.\n", service.Name, namespace)
+	return nil
+}
+
+func replaceService(client serving_v1alpha1_client.ServingV1alpha1Interface, service *serving_v1alpha1_api.Service, namespace string, out io.Writer) error {
+	existingService, err := client.Services(namespace).Get(service.Name, v1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	service.ResourceVersion = existingService.ResourceVersion
+	_, err = client.Services(namespace).Update(service)
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(out, "Service '%s' successfully replaced in namespace '%s'.\n", service.Name, namespace)
+	return nil
+}
+
+func serviceExists(client serving_v1alpha1_client.ServingV1alpha1Interface, name string, namespace string) (bool, error) {
+	_, err := client.Services(namespace).Get(name, v1.GetOptions{})
+	if api_errors.IsNotFound(err) {
+		return false, nil
+	}
+	if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+// Create service struct from provided options
+func constructService(cmd *cobra.Command, editFlags ConfigurationEditFlags, name string, namespace string) (*serving_v1alpha1_api.Service,
+	error) {
+
+	service := serving_v1alpha1_api.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+
+	// TODO: Should it always be `runLatest` ?
+	service.Spec.DeprecatedRunLatest = &serving_v1alpha1_api.RunLatestType{
+		Configuration: serving_v1alpha1_api.ConfigurationSpec{
+			DeprecatedRevisionTemplate: &serving_v1alpha1_api.RevisionTemplateSpec{
+				Spec: serving_v1alpha1_api.RevisionSpec{
+					DeprecatedContainer: &corev1.Container{},
+				},
+			},
+		},
+	}
+
+	if err := editFlags.ApplyService(&service); err != nil {
+		return nil, err
+	}
+
+	config, err := serving_lib.GetConfiguration(&service)
+	if err != nil {
+		return nil, err
+	}
+	err = editFlags.Apply(config, cmd)
+	if err != nil {
+		return nil, err
+	}
+	return &service, nil
+}
+
+func showUrl(client serving_v1alpha1_client.ServingV1alpha1Interface, serviceName string, namespace string, out io.Writer) error {
+	service, err := client.Services(namespace).Get(serviceName, v1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("cannot fetch service %s in namespace %s for extracting the URL", serviceName, namespace)
```

Comment:
Might read better with: "... to extract its URL" instead of "...for extracting the URL"